PyPDF2/generic.py

```diff
@@ -469,23 +478,22 @@ class NameObject(str, PdfObject):
     surfix = b_("/")
 
     def writeToStream(self, stream, encryption_key):
-        stream.write(b_(self))
+        stream.write(self.encode("utf8"))
 
     def readFromStream(stream, pdf):
         debug = False
-        if debug: print((stream.tell()))
+        if debug:
+            print((stream.tell()))
         name = stream.read(1)
         if name != NameObject.surfix:
             raise utils.PdfReadError("name read error")
-        name += utils.readUntilRegex(stream, NameObject.delimiterPattern,
-            ignore_eof=True)
-        if debug: print(name)
+        name += utils.readUntilRegex(
+            stream, NameObject.delimiterPattern, ignore_eof=True
+        )
+        if debug:
+            print(name)
         try:
-            try:
-                ret=name.decode('utf-8')
-            except (UnicodeEncodeError, UnicodeDecodeError) as e:
-                ret=name.decode('gbk')
-            return NameObject(ret)
+            return NameObject(name.decode("utf-8"))
```

Comment:
Ah, here is another change :-)

Reply:
Not sure this is matching table 3.3 rules...